sympy/functions/combinatorial/numbers.py

```diff
@@ -1752,29 +1752,28 @@ def _stirling2(n, k):
 
 
 def stirling(n, k, d=None, kind=2, signed=False):
-    r"""Return Stirling number `S(n, k)` of the first or second (default) kind.
+    r"""Return Stirling number $S(n, k)$ of the first or second (default) kind.
 
-    The sum of all Stirling numbers of the second kind for `k = 1`
-    through `n` is ``bell(n)``. The recurrence relationship for these numbers
+    The sum of all Stirling numbers of the second kind for $k = 1$
+    through $n$ is ``bell(n)``. The recurrence relationship for these numbers
     is:
 
     .. math :: {0 \brace 0} = 1; {n \brace 0} = {0 \brace k} = 0;
 
     .. math :: {{n+1} \brace k} = j {n \brace k} + {n \brace {k-1}}
 
-    where `j` is:
-        `n` for Stirling numbers of the first kind
-        `-n` for signed Stirling numbers of the first kind
-        `k` for Stirling numbers of the second kind
+    where $j$ is:
+        $n$ for Stirling numbers of the first kind,
+        $-n$ for signed Stirling numbers of the first kind,
+        $k$ for Stirling numbers of the second kind.
 
     The first kind of Stirling number counts the number of permutations of
     ``n`` distinct items that have ``k`` cycles; the second kind counts the
     ways in which ``n`` distinct items can be partitioned into ``k`` parts.
     If ``d`` is given, the "reduced Stirling number of the second kind" is
-    returned: ``S^{d}(n, k) = S(n - d + 1, k - d + 1)`` with ``n >= k >= d``.
-    (This counts the ways to partition ``n`` consecutive integers into
-    ``k`` groups with no pairwise difference less than ``d``. See example
-    below.)
+    returned: $S^{d}(n, k) = S(n - d + 1, k - d + 1)$ with $n \ge k \ge d$.
+    (This counts the ways to partition $n$ consecutive integers into $k$
+    groups with no pairwise difference less than $d$. See example below.)
```

Comment:
@sympy/gsod would be good to have some "style guide" guidance about where ` ``n`` ` and `$n$` are appropriate in this paragraph.

E.g., I don't think `S^{d}` should be typeset as code: its an exponent not some bitshift (or whatever it is that `^` does in Python).

Reply:
Yes, that is definitely something that should go in the style guide that @lglattly is writing. The general idea is that code (double backticks) should only be used for things that make sense literally in Python. So for instance obviously things like this shouldn't be code because `S^{d}` doesn't do anything meaningful in Python. But even something like just `S` is wrong to render as code, because `S` as a Python variable doesn't refer to the Stirling numbers. If it were `stirling` code would be correct. Sometimes the code and math would actually be the same, so you have to decide which is more appropriate from context. If the sentence in question is discussing mathematics, then LaTeX should be used, but if it is discussing details of the SymPy implementation specifically, then code should be used.

Reply:
And can the reader reasonably be expected to understand that  ` ``n`` ` and `$n$` in the same paragraph refer to the same object?  (see previous few lines to the my change above for an example).

Is so, I think our style-guide should say this explicitly.

Reply:
Thanks for bringing this up @cbm755! This is on my list as something we need to discuss and decide on, and I agree it should be stated clearly in the new style guide. I think it makes sense that if an object is both math and code, $$ should be used when referring to the math form and `` used when referring to the code form. I think a reader would understand this, but again this would be explained explicitly in the style guide to be safe.

Reply:
@cbm755 You should always imagine what you would do if the variable wasn't the same in code and LaTeX. For instance, what if instead of `n` it was a greek letter like alpha. In code, it should always be `alpha` and in math contexts it should always be `\alpha` (rendering as `α`). 

I think for any instance, it will always be one or the other, code or math. You just need to think about it from the context to know which it is. I don't know if there's a good one or two sentence rule of thumb that we can put in the style guide to help people differentiate it, beyond what I wrote above.

Reply:
> I think for any instance, it will always be one or the other, code or math. 

Well, its not so in the Stirling docs above...  assuming I've understood your comment correctly.

There are both ` ``n`` ` and `$n$` around in the same paragraph.  I wanted our future style guide to clarify to myself and future devs that this is ok (perhaps even expected?).

I agree any examples of this in our docs should include an ` ``alpha`` ` / `$\alpha$`.